### ISSUE TITLE
[Mono][Arm64]Improve the generated code for Vector128.Equals* on Arm64

### ISSUE THREAD

**Issue by fanyang-mono** [dotnet/runtime#69325]
Perf_VectorOf<T>.Equals* and Perf_VectorOf<T>.Greater/Less* microbenchmarks are much slower when running on Mono comparing to CoreCLR. We should try to optimize `OP_XEQUAL` in mini-llvm.c for Arm64, while not regressing it for amd64.

Currently, the numbers are
| Benchmark name                                                                                        | CoreCLR (ns) | Mono (ns) |
| --------------------------------------------------------------------  |:-------------:| -----------:|
|System.Numerics.Tests.Perf_VectorOf<UInt32>.EqualsAllBenchmark | 0 | 13.51 |
|System.Numerics.Tests.Perf_VectorOf<Int64>.InequalityOperatorBenchmark | 0 | 13.12 |
|System.Numerics.Tests.Perf_VectorOf<SByte>.EqualsAnyBenchmark | 0 | 15.82 |
|System.Numerics.Tests.Perf_VectorOf<SByte>.GreaterThanAllBenchmark | 0 | 15.83 |
|System.Numerics.Tests.Perf_VectorOf<UInt32>.LessThanOrEqualAllBenchmark | 0 | 13.52 |

Use this simple app as an example,
```
using System;
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.Arm;
using System.Runtime.CompilerServices;
using System.Numerics;

namespace HelloWorld
{
    internal class Program
    {
        [MethodImpl(MethodImplOptions.NoInlining)]
        private static bool test(Vector128<int> a, Vector128<int> b)
        {
            return Vector128.EqualsAll(a,b);
        }

        private static void Main(string[] args)
        {
            Vector128<int> A = Vector128.Create((int)3.1);
            Vector128<int> B = Vector128.Create((int)5.7);

            var result = test(A, B);
            Console.WriteLine(result);
        }
    }
}
```

On Arm64, the Mono LLVM JIT generated assembly code for `test` method is
```
0000000000000000 <.text>:
   0:	9e670000 	fmov	d0, x0
   4:	9e670021 	fmov	d1, x1
   8:	9e670042 	fmov	d2, x2
   c:	6e180420 	mov	v0.d[1], v1.d[0]
  10:	9e670061 	fmov	d1, x3
  14:	6e180422 	mov	v2.d[1], v1.d[0]
  18:	6ea28c00 	cmeq	v0.4s, v0.4s, v2.4s
  1c:	4e040401 	dup	v1.4s, v0.s[0]
  20:	6e014001 	ext	v1.16b, v0.16b, v1.16b, #8
  24:	4e201c20 	and	v0.16b, v1.16b, v0.16b
  28:	4e0c0401 	dup	v1.4s, v0.s[1]
  2c:	4e211c00 	and	v0.16b, v0.16b, v1.16b
  30:	1e260008 	fmov	w8, s0
  34:	7100011f 	cmp	w8, #0x0
  38:	1a9f07e0 	cset	w0, ne  // ne = any
  3c:	d65f03c0 	ret
```

On Arm64, the CoreCLR generated assembly code for `test` method is (Tier-0 compilation)
```
G_M6901_IG01:              ;; offset=0000H
        A9BD7BFD          stp     fp, lr, [sp,#-48]!
        910003FD          mov     fp, sp
        3D800BA0          str     q0, [fp,#32]
        3D8007A1          str     q1, [fp,#16]  
                                                ;; size=16 bbWeight=1    PerfScore 3.50
G_M6901_IG02:              ;; offset=0010H
        3DC00BB0          ldr     q16, [fp,#32]
        3DC007B1          ldr     q17, [fp,#16]
        6EB18E10          cmeq    v16.4s, v16.4s, v17.4s
        6E31AA10          uminv   b16, v16.16b
        0E013E00          umov    w0, v16.b[0]
        7100001F          cmp     w0, #0
        9A9F07E0          cset    x0, ne
                                                ;; size=28 bbWeight=1    PerfScore 10.00
G_M6901_IG03:              ;; offset=002CH
        A8C37BFD          ldp     fp, lr, [sp],#48
        D65F03C0          ret     lr

```

On Amd64, the Mono LLVM JIT generated assembly code for `test` method is
```
0000000000000000	vmovq	%rsi, %xmm0
0000000000000005	vmovq	%rdi, %xmm1
000000000000000a	vpunpcklqdq	%xmm0, %xmm1, %xmm0
000000000000000e	vmovq	%rcx, %xmm1
0000000000000013	vmovq	%rdx, %xmm2
0000000000000018	vpunpcklqdq	%xmm1, %xmm2, %xmm1
000000000000001c	vpcmpeqd	%xmm1, %xmm0, %xmm0
0000000000000020	vmovmskps	%xmm0, %eax
0000000000000024	cmpl	$0xf, %eax
0000000000000027	sete	%al
000000000000002a	retq
```

**Comment by fanyang-mono** [comment]
I need more time to get the complete fix for this, moving it to 8.0

**Comment by TamarChristinaArm** [comment]
Hi, just a head's up, the implementation for these builtins were changed into better ones for CoreCLR See #75864 , implementation should be easier to do, you might want to implement that one instead.
